app/models/middleware_server_performance.rb

```diff
@@ -16,7 +16,11 @@ def self.build_results_for_report_middleware(options)
     start_time, end_time, interval = parse_time_interval(options)
     results = []
     MiddlewareServer.find_each do |ms|
-      raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
+      begin
+        raw_stats = fetch_raw_stats(ms, start_time, end_time, interval)
```

Comment:
From a provider pluggability view, you should define an Exception in MiddlewareServerPerformance that is used to do something, like throttling the update interval.

Then in the  MiddlewareServer subclass (Hawkular::MiddlewareServer) translate the known exceptions from external gems into the one defined here.

That being said, maybe 

> It probably makes sense to "just abort" when the issue is really not inside the Hawkular code or responsibility.

See my throttling comment above. You only want to handle exceptions here, if you really want to act upon them. And the action should be defined here and therefore the exception too.